File: 1267-count-servers-that-communicate/1267-count-servers-that-communicate.cpp
class Solution {
public:
    int countServers(vector<vector<int>>& grid) {
        int gridSize = grid.size();
        int col = grid[0].size();
        int *gridColSize = &col;
        int **seen = (int**)calloc(sizeof(int*), gridSize);
    for (int i = 0; i < gridSize; i++){
        seen[i] = (int*)calloc(sizeof(int), (*gridColSize));
    }
    int connected = 0;

    for (int i = 0; i < gridSize; i++){
        int frow = 0, fcol = 0, comAll = 0, comSeen = 0;
        for (int j = 0; j < (*gridColSize); j++){
            if (grid[i][j]){
                if (!comAll){
                    frow = i; fcol = j;
                }
                comAll += 1;
                comSeen += ((seen[i][j] >= 1)? 1:0);
                seen[i][j] += 1;
            }
        }
        if (comAll == 1){
            seen[frow][fcol] -= 1;
        }
        else if (comAll > 1){
            connected += comAll-comSeen;
        }
    }

    for (int j = 0; j < (*gridColSize); j++){
        int frow = 0, fcol = 0, comAll = 0, comSeen = 0;
        for (int i = 0; i < gridSize; i++){
            if (grid[i][j]){
                if (!comAll){
                    frow = i; fcol = j;
                }
                comAll += 1;
                comSeen += ((seen[i][j] >= 1) ? 1:0);
                seen[i][j] += 1;
            }
        }
        if (comAll == 1){
            seen[frow][fcol] -= 1;
        }
        else if (comAll > 1){
            connected += comAll-comSeen;
        }
    }

    return connected;
    }
};

File: 1267-count-servers-that-communicate/1267-count-servers-that-communicate.py
class Solution:
    def countServers(self, grid: List[List[int]]) -> int:
        seen = [[0]*len(grid[0]) for _ in range(len(grid))]
        connected = 0
        # Check rows first
        for i in range(len(grid)):
            # First two are to point out position of first computer in current row
            # comAll is count of all computer in current row
            # comSeen is count of all computer in current row that are previously seen
            # So, current row will have (comAll-comSeen) new computer connected
            frow, fcol, comAll, comSeen = None, None, 0, 0
            for j in range(len(grid[0])):
                if grid[i][j]:
                    if frow is None and fcol is None:
                        frow, fcol = i, j
                    comAll += 1
                    comSeen += 1 if seen[i][j] else 0
                    seen[i][j] += 1
            
            # IF there is only one computer in the whole row, then we are going to make it unseen
            if comAll == 1:
                seen[frow][fcol] -= 1
            elif comAll > 1:
                connected += comAll - comSeen
        
        # Now do the same for all columns
        for j in range(len(grid[0])):
            frow, fcol, comAll, comSeen = None, None, 0, 0
            for i in range(len(grid)):
                if grid[i][j]:
                    if frow is None and fcol is None:
                        frow, fcol = i, j
                    comAll += 1
                    comSeen += 1 if seen[i][j] else 0
                    seen[i][j] += 1
            if comAll == 1:
                seen[frow][fcol] -= 1
            elif comAll > 1:
                connected += comAll - comSeen
        
        return connected